llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -1713,6 +1713,15 @@ LegalizerHelper::LegalizeResult LegalizerHelper::narrowScalar(MachineInstr &MI,
     MI.eraseFromParent();
     return Legalized;
   }
+  case TargetOpcode::G_SPLAT_VECTOR: {
+    if (TypeIdx != 1)
+      return UnableToLegalize;
+
+    Observer.changingInstr(MI);
+    narrowScalarSrc(MI, NarrowTy, 1);
```

Comment:
This silently drops the upper bits of the scalar and creates a splat vector where the source scalar type is smaller than the vector element type.